sdk/nodejs/cmd/run/run.ts

```diff
@@ -215,6 +216,28 @@ ${defaultMessage}`);
 
     programStarted();
 
+    if (isQueryModeEnabled()) {
+        log.debug(`Running program '${program}' in pwd '${process.cwd()}' w/ args: ${programArgs}`);
+        try {
+            require(program);
+        } catch (e) {
+            // User JavaScript can throw anything, so if it's not an Error it's definitely
+            // not something we want to catch up here.
+            if (!(e instanceof Error)) {
+                throw e;
+            }
+
+            // Give a better error message, if we can.
+            const errorCode = (<any>e).code;
+            if (errorCode === "MODULE_NOT_FOUND") {
+                reportModuleLoadFailure(program, e);
+            }
+
+            throw e;
```

Comment:
Should we extract this out instead of copy/paste?

Reply:
Yes.